src/api/app/models/workflow/step.rb

```diff
@@ -25,4 +25,99 @@ def source_project_name
   def target_project_name
     "home:#{@token.user.login}:#{source_project_name}:PR-#{scm_extractor_payload[:pr_number]}"
   end
+
+  def source_package_name
```

Comment:
I don't think all steps really need this. So far, it's only 2 out of 3 steps: `link_package` and `branch_package`. I would prefer to keep the duplicated code in 2 steps and extract this if we happen to need the same code in a third step. This is a good example of the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)).

Reply:
`Workflow::Step` might not be the right place, I agree. But for *sure* we should not have this complicated code twice.

Reply:
@dmarcoux , we can understand your point, but keep in mind it's not only the `source_package_name` method, there are more than 10 methods that are identical in both `LinkPackageStep` and `BranchPackageStep`.
Take also into account that, at the time we implemented this, we had 2 out of 2 steps containing the same code.

We think all of this will change as we are introducing new steps. We prefer to avoid duplication now and plan better refactoring for the near future. Anyway, we are going to move the extraction to an independent PR (which is in [this commit](https://github.com/openSUSE/open-build-service/pull/11413/commits/acadd11f6be3dcd1bbaf7761dd108be33a1fda08)).

Reply:
This code has been moved to a new PR: #11433